stdlib/universe/difference.go

```diff
@@ -344,8 +371,15 @@ func (d *difference) updateInt(v int64, valid bool) (int64, bool) {
 		d.valid = true
 		return 0, false
 	}
-	if diff := v - prev; diff >= 0 || !d.nonNegative {
+	diff := v - prev
+	if diff >= 0 || !d.nonNegative {
 		return diff, true
+	} else if diff < 0 && d.nonNegative && d.initialZero {
```

Comment:
The `diff < 0` is unnecessary as it's redundant. Since the previous condition checks using `||`, then diff must be less than zero.

It also seems to me like the line on 381 doesn't make sense to me. In the case that the next value is less than the initial value of zero, I think we should go with the default behavior where we return `0, false` and output a null column.

So I think this condition should be changed to: `d.nonNegative && d.initialZero && v >= 0` and then you can just `return v, true` and fallthrough to the next behavior.

Reply:
_"The diff < 0 is unnecessary as it's redundant. Since the previous condition checks using ||, then diff must be less than zero."_
Adding `diff < 0` was intentional for the code readability, but if it seems obvious, I will remove that. 

_It also seems to me like the line on 381 doesn't make sense to me. In the case that the next value is less than the initial value of zero, I think we should go with the default behavior where we return 0, false and output a null column._
I have implemented this as per the requirement in the [issue](https://github.com/influxdata/flux/issues/4098). I have used the exact same table as mentioned in the issue in the flux [test](https://github.com/influxdata/flux/blob/24dbe5474595fb8420990b80c49f8debfa4772e0/stdlib/universe/increase_test.flux#L69). I can implement it in the way you suggested. We can discuss it with Scott if required.

Reply:
If this is a requirement or the behavior of how prometheus's increase function works, I think I would prefer whatever that output is. I'm not sure I see this requirement in the comment you linked. It seems like @sanderson is talking about the first row while I'm referring to the reset behavior when the reset value is less than zero. As an example of input: `[10, 20, -10, 20]`. Since we're saying the initial value is zero when a reset happens, but the new value is less than zero, but we said non negative is true, we shouldn't produce negative values. But, producing a value of zero seems wrong to me since that's not really the math that was used. It seems safest to me to just output null in this case.

Reply:
That bit was not clear to me either so I have used the query provided in the issue. Below is the excerpt from the query. 
`|> map(fn: (r) => ({ r with diff: if (not exists r.diff and r._value >= 0) then r._value else if (not exists r.diff and r._value < 0) then float(v: 0) else r.diff }))`
As suggested, I have updated the code to output null in this case.